velox/exec/PartitionedOutput.h

```diff
@@ -51,11 +51,13 @@ class Destination {
       const std::vector<vector_size_t>& sizes,
       const RowVectorPtr& output,
       PartitionedOutputBufferManager& bufferManager,
+      std::function<void()> bufferReleaseFn,
```

Comment:
Good point. But the client here is the partition output operator which buffers data and flush into the system wide buffer manager as one serialized io buffer at a time once the buffered data reaches to a preset limit or on output operator close. The output operator's lifecycle (client) is bound to the task but the created io buffers are stored in output buffer manager and are consumed by external component like HTTP server in Prestimisso. So the io buffer could outlive the client that generates the buffers. It might be a little bit heavy to hold a task reference on each buffer input to the partition output operator and we don't have such abstraction yet. This PR just extends the current iobuf ownership handling a bit by holding an additional reference on the task.